Kernel/Process.cpp

```diff
@@ -4290,7 +4290,7 @@ int Process::sys$setkeymap(const Syscall::SC_setkeymap_params* user_params)
     if (!is_superuser())
         return -EPERM;
 
-    REQUIRE_NO_PROMISES;
+    REQUIRE_PROMISE(stdio);
```

Comment:
Disagree. Changing the system keymap is not something you'd want a process that has only promised stdio to be able to do.

Generally, stdio means only doing I/O on the file descriptors you already have opened (capability-style), manipulating your own memory mappings, and some other basic stuff like getting your PID. Here's a simple test you could use to decide if some functionality fits into stdio: can it be used to impact the rest of the system in any way, not counting any pre-opened files? If so, it's a bad fit for stdio.

Reply:
Hi Sergey,
When program sets pledge("x") REQUIRE_NO_PROMISES raise error. For me the best is stdio :)

Reply:
That is the point of `REQUIRE_NO_PROMISES`, yes :) we really don't want sandboxed programs messing with system keymap, do we?

Reply:
I roll back it to `REQUIRE_NO_PROMISESI`